### ISSUE TITLE
Testing Generated QR Codes on Banking Apps

### ISSUE THREAD

**Issue by xseman** [xseman/bysquare#42]
I need help testing QR codes created with this library on various Slovak banking apps. I want to ensure these codes work properly with different types of data.

Unfortunately, some banking apps might not correctly read QR codes. For example, Tatrabanka's app can't handle QR codes with multiple payment options. While I can help identify issues with the QR codes themselves, I can only test one banking app at a time right now, so I can't check every app out there.

Here is a list of banks that I want to verify for compatibility:

- [ ] 365.bank
- [ ] ČSOB
- [ ] Poštová banka
- [ ] Prima banka
- [ ] Privatbanka
- [ ] Prvá stavebná sporiteľňa
- [ ] Raiffeisen Bank
- [ ] Slovenská sporiteľňa
- [ ] Slovenská záručná a rozvojová banka
- [ ] Tatra banka
- [ ] Všeobecná úverová banka
- [ ] Wüstenrot stavebná sporiteľňa

All I need is the input used for `encode()`, the resulting output, and a screenshot of the banking app showing that the QR code was recognized. To generate QR code from string use any online app, for example https://www.qr-code-generator.com/

**Comment by xseman** [comment]
## Tatra banka

**Input**

```json
{
	"invoiceId": "random-id",
	"payments": [
		{
			"type": 1,
			"amount": 100.0,
			"bankAccounts": [
				{ 
					"iban": "SK9611000000002918599669"
				}
			],
			"currencyCode": "EUR",
			"variableSymbol": "hello",
			"paymentNote": "world"
		}
	]
}
```

**Output**

```plain
0004O0001I37SM453J3K7EO06KEMO475RRD60TPLKIQTU9RD5P2BF2CTBLM60C155QV8H57OVGBK4S7MPA3SR7SJQU1B0UV92NFJ702QKDHH9P4FVVVU8F0000
```

**Screenshot**

<img src="https://github.com/user-attachments/assets/79871a93-cf75-4548-8696-235051de20ea" width="200">

**Comment by LukasBicus** [comment]
## Fio
| Payment Type  | Status                   | QR Code                                                                                                  |
| ------------- | ------------------------ |----------------------------------------------------------------------------------------------------------|
| PaymentOrder  | :x:       | <img src="https://github.com/user-attachments/assets/ee139392-5676-4059-996e-decf977683ab" width="200">  |
| StandingOrder | :heavy_multiplication_x: | <img src="" width="200">                                                                                 |
| DirectDebit   | :heavy_multiplication_x: | <img src="" width="200">                                                                                 |

**Comment by LukasBicus** [comment]
## mBank

| Payment Type  | Status                   | QR Code                                                                                                  |
| ------------- | ------------------------ |----------------------------------------------------------------------------------------------------------|
| PaymentOrder  | :x:       | <img src="https://github.com/user-attachments/assets/b836d699-1b95-4aa2-8107-9f70e51998e0" width="200">  |
| StandingOrder | :heavy_multiplication_x: | <img src="" width="200">                                                                                 |
| DirectDebit   | :heavy_multiplication_x: | <img src="" width="200">                                                                                 |

**Comment by LukasBicus** [comment]
@xseman - I tested the QR code from the preact demo too. It works with variable `123`. It doesn't work with variable `hello`.

**Comment by xseman** [comment]
@LukasBicus you're correct, there's an issue with the TatraBanka application, it accepts strings, but when entered manually, it only allows numbers. I'll update the issue with a new QR code containing just numbers.

**Comment by michalvankodev** [comment]
## Slovenska sporitelna (George 24.26.12-google)

|       PaymentOrder       |      StandingOrder       |       DirectDebit        |
| :----------------------: | :----------------------: | :----------------------: |
| <img src="https://github.com/user-attachments/assets/49bce0a4-0221-4e6c-8181-4996fc058e4d" width="200"> | <img src="https://github.com/user-attachments/assets/f9943196-9ec0-4120-9be9-87ec681d9edf" width="200"> | <img src="https://github.com/user-attachments/assets/d70ec5ba-e2a3-4143-ae62-787b6b9312f2" width="200"> |

**Comment by michalvankodev** [comment]
![Screenshot_2024-08-27-11-31-55-657_bank sk365 app]()
## 365.bank (27.9.5)

|       PaymentOrder       |      StandingOrder       |       DirectDebit        |
| :----------------------: | :----------------------: | :----------------------: |
| <img src="https://github.com/user-attachments/assets/7451bb9f-4f18-4933-9c85-3ec52d1eab29" width="200"> |  |  |

App crashes when trying to scan standing or direct order.

**Comment by dderevjanik** [comment]
@xseman tested on VUB mobile app (old one). Standing Order and Direct Debit are not working (incorrect QR code message)

|       PaymentOrder       |      StandingOrder       |       DirectDebit        |
| :----------------------: | :----------------------: | :----------------------: |
| ![image](https://github.com/user-attachments/assets/a790dc6e-a0ef-480c-8769-6b0e97867971) | ❌ | ❌ |

**Comment by Dodkes** [comment]
## UniCredit Bank (v5.45.49)

|       PaymentOrder       |      StandingOrder       |       DirectDebit        |
| :----------------------: | :----------------------: | :----------------------: |
| <img src="https://github.com/user-attachments/assets/54d34328-c216-4036-b692-01422f1d25a1" width="200"> | <img src="https://github.com/user-attachments/assets/ebab6625-bf74-4289-b32f-ad379fcd120c" width="200"> | <img src="https://github.com/user-attachments/assets/73117f5b-cccf-4294-8962-bf21e02774c9" width="200"> |

**Comment by xseman** [comment]
Thanks a lot for the feedback, @tukusejssirs! 👍👍

**Comment by xseman** [comment]
## PrmaBanka (android v5.56.1)

|       PaymentOrder       |      StandingOrder       |       DirectDebit        |
| :----------------------: | :----------------------: | :----------------------: |
| <img src="https://github.com/user-attachments/assets/7d713d56-283e-482e-9bb8-6288c289be0c" width="200"> | <img src="https://github.com/user-attachments/assets/f1e6e1b7-dedf-4cad-b44b-17b20b8fc844" width="200"> | <img src="https://github.com/user-attachments/assets/cf159999-b7ef-4200-b92e-7f92dc4f764e" width="200"> |

**Comment by xseman** [comment]
Uzatváram, vznikol nový issiue na trackovanie bank https://github.com/xseman/bysquare/issues/133
